test-resources-plugin/src/main/java/io/micronaut/gradle/testresources/MicronautTestResourcesConsumerPlugin.java

```diff
@@ -39,23 +37,36 @@ public class MicronautTestResourcesConsumerPlugin implements Plugin<Project> {
     public void apply(Project project) {
         PluginManager pluginManager = project.getPluginManager();
         pluginManager.apply(MicronautBasePlugin.class);
-        TestResourcesConsumerConfiguration testResourcesExtension = createTestResourcesExtension(project);
-        pluginManager.withPlugin("io.micronaut.component", unused -> {
-            project.afterEvaluate(p -> p.getConfigurations().all(cnf -> configureDependencies(p, testResourcesExtension, project.getDependencies(), cnf)));
+        Configuration testResourcesConfiguration = createTestResourcesExtension(project);
+        pluginManager.withPlugin("io.micronaut.component", unused ->
+                project.afterEvaluate(p -> p.getConfigurations().all(cnf -> configureDependencies(testResourcesConfiguration, cnf)))
+        );
+    }
+
+    private Configuration createTestResourcesExtension(Project project) {
+        return project.getConfigurations().create(MicronautTestResourcesPlugin.TESTRESOURCES_CONFIGURATION, conf -> {
+            conf.setCanBeConsumed(false);
+            conf.setCanBeResolved(false);
+            conf.setDescription("Used to declare projects which provide test resources");
+            conf.getDependencies().whenObjectAdded(dep -> {
+                if (dep instanceof ProjectDependency) {
+                    ProjectDependency projectDependency = (ProjectDependency) dep;
+                    projectDependency.attributes(attrs -> configureUsageAttribute(project, attrs));
```

Comment:
I thought the attributes should be configured at the configuration level

Reply:
They could, for configurations which can be resolved. However if we put it on the configuration, then _all_ dependencies will be resolved with those attributes. This is not what we want here, since only the test resources dependencies need to resolve the test resources variant.